### PR TITLE
Fixes build issue on netlify

### LINKED ISSUE #235: Website builds aren't working
Netlify's production builds of spiffe.io are not working.
See logs: https://app.netlify.com/sites/spiffe/deploys/62337f374e585f229a8c01cb

```
3:34:57 PM: Installing dependencies from Pipfile
3:34:58 PM: Warning: Python 3.7 was not found on your system...
3:34:58 PM: Neither 'pyenv' nor 'asdf' could be found to install Python.
```

### PR DESCRIPTION
**Pull request by mchurichi**
Fixes #235.

Changes:
- Upgrades to use Netlify image with Python 3.8
- Adds support for `branch` config in the external.yaml file, to allow pulling from different branches (some spiffe/spire repos were migrated to use `main`, some others still use `master`)
- Adds pyenv to install Python version in local development environment
- Pins node js version

### CODE REVIEW

**Comment by mchurichi** [comment]
@sanderson042 @Andres-GC would any of you folks mind taking a quick look at this?

File: Dockerfile.dev [hunk 1]
```diff
@@ -6,23 +6,39 @@ ENV PYTHONDONTWRITEBYTECODE 1
...
 
 # Installs Pyenv
-ENV PATH="/root/.pyenv/bin:${PATH}"
+ENV PATH="/home/appuser/.pyenv/bin:/home/appuser/.local/bin:${PATH}"
 RUN curl https://pyenv.run | bash
+RUN export PATH="$(pyenv root)/shims/python:$PATH"
 RUN echo 'eval "$(pyenv init -)"' >> ~/.bashrc
+RUN echo 'eval "$(pyenv init --path)"' >> ~/.bashrc
+# RUN eval "$(pyenv init - sh)"
+# RUN eval "$(pyenv init --path)"
```
Comment (Andres-GC) [inline Dockerfile.dev:26]: I assume you only need one of the first to lines, right?
If possible, let's remove the commented lines here please.

**Review by Andres-GC — APPROVED** [review]
Thanks Max for this.
I just left a minor comment to remove a comment line. Apart from it, it is ok.

File: Dockerfile.dev (hunk 1 above)
Comment (mchurichi) [reply Dockerfile.dev:26]: indeed, thanks!
will remove the commented lines

**Pull request merged**